### DISCUSSION TITLE
Missing hover effects

### DISCUSSION THREAD

**Discussion by Princesseuh** [withastro/starlight#1912] [Feature Requests]
### What version of `starlight` are you using?

0.23.1

### What version of `astro` are you using?

4.8.6

### What package manager are you using?

pnpm

### What operating system are you using?

macOS

### What browser are you using?

Arc

### Describe the Bug

This is probably not a bug, but it kinda feels like one since it's inconsistent with other elements.

In the default theme, many elements are missing an hover effect, for instance, hovering the Get Started button on https://starlight.astro.build/ gives no feedback apart from the cursor changing, same for the "View on GitHub" link next to it.

On Starlight websites without an image logo, the logo also does not change visually when hovered. This is visible on the default starter, since it has a text logo.

Less important, but headers in the sidebar (the foldable ones) also have no hover effect apart from the cursor.

### Link to Minimal Reproducible Example

https://stackblitz.com/github/withastro/starlight/tree/main/examples/basics

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Thanks for the issue! Converted this to a feature request as I agree that it probably isn’t strictly speaking a “bug”.

Some reference links that might be helpful for folks:

| Design system | Has hover style |
|---------------|:---------------:|
| [Base Web (Uber)](https://baseweb.design/components/button/) | ✅ |
| [Bootstrap](https://getbootstrap.com/docs/4.3/components/buttons/) | ✅ |
| [Carbon (IBM)](https://carbondesignsystem.com/components/button/usage/) | ✅ |
| [GOV.UK DS](https://design-system.service.gov.uk/components/button/) | ✅ |
| [Primer (GitHub)](https://primer.style/components/button#variants) | ✅ |
| [USWDS](https://designsystem.digital.gov/components/button/) | ✅ |

Pretty universally these use a very subtle lightening or darkening of the button background colour on hover.
